### ISSUE TITLE
Installation Failure Mode: Remove `augur.config.json` on failed `make install-dev`

### ISSUE THREAD

**Issue by sgoggins** [chaoss/augur#680]
**Describe the bug:**
If my `make install-dev` fails for any reason, it will continue to fail until I `rm augur.config.json`, because whatever I got wrong remains in the file and I am not prompted again. 

**To Reproduce:**
1. run `'make install-dev`
2. provide bad database host, or bad repo path info
3. fix the issue
4. try again: `make install-dev`
5. See error

**Expected behavior:**
If I do something that causes my installation to fail, it is not clear that I have to delete my `augur.config.json` to try again. This make be documented, but I think this is something we should clean up for a user on a `make install-dev`. 

**Desktop (please complete the following information):**
 - Ubuntu 18

**Additional context**
`dev branch`

**Comment by ccarterlandis** [comment]
I don't know what "error" you're talking about. Can you please provide log files if you have any?

If you're referring to the fact that the install script doesn't prompt you to generate another config when you run it again, this is expected behavior under the requirements which were originally given to me when I wrote the script. That feature is easily changed, but it exists because I thought that's what we wanted.

**Comment by sgoggins** [comment]
I agree. during testing yesterday I had a number of self induced head wounds and wanted to start over. Whenever I started over, it failed at the same place. Shortly, I realized I needed to delete the `augur.config.json` ... but I've been doing this for over 3 years, so I figure ... Removing the augur.config.json if the user doesn't at least get past the front end installation for now seems like its more what a new installer will expect. After having done it. Which is distinct from making it up in my head. :)

**Comment by ccarterlandis** [comment]
If I understand what you're saying correctly (which I hope I do), I disagree. I think that each piece of the install process is a separate task, and if one fails, then I don't see why we would need to undo the previous sections if they worked. The only scenario where I think it would be appropriate to delete the config and ask the user to do it again is if the config was somehow formed incorrectly and needed to be rebuilt. 

If it were me installing Augur for the first time (oh how long ago that was 😂), if I successfully generated my config file and then it got deleted because the frontend installation process (which is in my mind completely unrelated to the config) failed to complete, I would be very confused and probably a bit annoyed that I had to type everything out again.

I would rather extract the config setup script into its own `make` command (which would be very easy) to allow people to regenerate their config if they needed. In my mind, if we did this, there should still be some sort of check for a config file (like `Hey, we noticed you already have a config file - are you sure you want to run this script and delete it?`), but it would serve the purpose of allowing people to edit mistakes to their config after the fact without introducing an artificial dependency on some other portion of the build process.
